multicloud/src/main/java/de/uniulm/omi/cloudiator/sword/multicloud/pricing/aws/AWSPricingSupplier.java

```diff
@@ -0,0 +1,204 @@
+                LOGGER.info(String.format("AWSPricingPack no: %d", ++awsPricingPackCounter));
+            }
+            else {
+                LOGGER.info("AWSPricingPack is empty");
+            }*/
+
+            for(AWSProductAndTerms awsProductAndTerms : awsPricingPack.getAwsProductAndTerms()) {
+                if(!productFamiliesFilter.contains(awsProductAndTerms.product.productFamily)
+                        || !tenanciesFilter.contains(awsProductAndTerms.product.attributes.get("tenancy"))
+                        || !capacityStatusesFilter.contains(awsProductAndTerms.product.attributes.get("capacitystatus"))
+                        || !awsProductAndTerms.product.attributes.get("operation").startsWith(operationsPatternFilter)) {
+                    continue;
+                }
+
+                OperatingSystemArchitecture operatingSystemArchitecture;
+                switch (awsProductAndTerms.product.attributes.get("processorArchitecture")) {
+                    case "64-bit": {
+                        operatingSystemArchitecture = OperatingSystemArchitecture.AMD64;
+                        break;
+                    }
+                    case "32-bit or 64-bit": {
+                        operatingSystemArchitecture = OperatingSystemArchitecture.I368;
+                        break;
+                    }
+                    default: {
+                        operatingSystemArchitecture = OperatingSystemArchitecture.UNKNOWN;
+                        break;
+                    }
+                }
+
+                OperatingSystemFamily operatingSystemFamily;
+                switch (awsProductAndTerms.product.attributes.get("operatingSystem")) {
+                    case "Linux": {
+                        operatingSystemFamily = OperatingSystemFamily.UBUNTU; // UBUNTU as a generalization of Linux OS distributions
+                        break;
+                    }
+                    case "RHEL": {
+                        operatingSystemFamily = OperatingSystemFamily.RHEL;
+                        break;
+                    }
+                    case "SUSE": {
+                        operatingSystemFamily = OperatingSystemFamily.SUSE;
+                        break;
+                    }
+                    case "Windows": {
+                        operatingSystemFamily = OperatingSystemFamily.WINDOWS;
+                        break;
+                    }
+                    default: {
+                        operatingSystemFamily = OperatingSystemFamily.UNKNOWN;
+                        break;
+                    }
+                }
+
+                OperatingSystemBuilder operatingSystemBuilder = OperatingSystemBuilder.newBuilder()
+                        .architecture(operatingSystemArchitecture)
+                        .family(operatingSystemFamily)
+                        .version(OperatingSystemVersions.unknown()); // AWS Pricing API exposes no information about version
```

Comment:
.build() could be invoked at the end of this statement.

Reply:
leaving as is